### DISCUSSION TITLE
What to do if homedir is a git repo?

### DISCUSSION THREAD

**Discussion by kaihendry** [sharkdp/fd#1299] [Q&A]
The way I manage my $HOME is via git and subsequently unless I'm in a git repo I think https://github.com/kaihendry/dotfiles/blob/master/.gitignore is read by fd and I'd rather it didn't.

I tried

```
❯ cat .fdignore
~/.git
~/.gitignore
```

but it doesn't work. 😬

I don't want to pass `fd -HI` everytime I use it!

**Comment by wellkilo** [comment]
Your .fdignore patterns do not work because ignore files are relative-pattern files, not shell path expressions: ~ is not expanded there. Also, disabling all ignore processing with -I would disable .fdignore too.

Use the narrower VCS-ignore switch and keep hidden-file filtering enabled:

    fd --no-ignore-vcs

That stops reading .gitignore/.git/info/exclude/global Git excludes while fd still hides dotfiles and .git by default. If you also want selected hidden paths, add -H and explicitly exclude .git:

    fd --no-ignore-vcs --hidden --exclude .git

To make this permanent, use a shell alias/function:

    alias fd='fd --no-ignore-vcs'

or configure the equivalent default options in your preferred wrapper. A global fd ignore file can exclude .git when -H is used; on Linux/macOS it is normally ~/.config/fd/ignore and should contain:

    .git/

not ~/.git.

The current README documents hidden handling, VCS ignore behavior, --exclude, and the global ignore path: [Hidden and ignored files](https://github.com/sharkdp/fd/blob/9e8927e8f3ee7bb699df55cb0b440f848dce6857/README.md#hidden-and-ignored-files), [global ignore](https://github.com/sharkdp/fd/blob/9e8927e8f3ee7bb699df55cb0b440f848dce6857/README.md#excluding-specific-files-or-directories).

**Reply by tmccombs** [reply]
There is also the `--no-ignore-parent` option, which tells fd not to respect ignore files (including .gitignore) higher up in the file hierarchy.

**Reply by wellkilo** [reply]
Good addition — --no-ignore-parent is the narrower option for this exact case.

It ignores .gitignore, .ignore, and .fdignore files inherited from directories above the search root, while preserving ignore files at or below the directory being searched. My earlier --no-ignore-vcs suggestion disables VCS ignore handling more broadly.

So when the unwanted rules come specifically from the home directory being a parent Git repository, the preferred command is:

```bash
fd --no-ignore-parent
```

Thanks for pointing out the more precise flag.
